spec.md

```diff
@@ -586,18 +589,68 @@ schema being used.
           "40 concurrent connections"
         ]
       }
+    }, {
+      "name": "fake-deprecated-plan",
+      "id": "009aba-XXXX-XXXX-XXXX-dace631cd648",
+      "description": "This is deprecated.",
+      "free": false,
+      "deprecation": {
+        "description": "Don't use this anymore.",
+        "since": "2018-04-20T17:00:00+00:00",
+        "eol": "2020-02-28T08:00:00+00:00",
+        "alternatives": [{
+            "description": "Use this plan instead.",
+            "plan_id": "0f4008b5-XXXX-XXXX-XXXX-dace631cd648"
+          }
+        ]
+      }
     }]
   }]
 }
 ```
 
+##### Deprecation Object 
+
+| Response Field | Type | Description |
+| --- | --- | --- |
+| description | string | A short description of the deprecation. |
+| since | date string | Date and time when the Service Broker deprecated the service or plan. If specified, it is expected to be an ISO 8601 formatted date and time, with time zone. |
+| eol | date string | Date and time when the Service Broker expects to end-of-life (remove) the service or plan from the catalog. If specified, it is expected to be an ISO 8601 formatted date and time, with time zone. |
+| alternatives | array of [CatalogObjectId](#catalog-object-id-object) objects | An array of services or plans that the Service Broker considers to be good alternatives. |
+
+\* Fields with an asterisk are REQUIRED.
+
+##### Catalog Object Id Object 
+
+| Response Field | Type | Description |
+| --- | --- | --- |
+| description | string | A short description to provide contextual information about this catalog object reference. |
+| service_id | string | If present, it MUST be the ID of a service. |
+| plan_id | string | If present, it MUST be the ID of a plan. |
+
+\* Fields with an asterisk are REQUIRED.
 
 ### Adding a Service Broker to the Platform
 
 After implementing the first endpoint `GET /v2/catalog` documented
 [above](#catalog-management), the Service Broker will need to be registered
 with your Platform to make your services and plans available to end users.
 
+### Deprecating services and plans
+
+A Service Broker can signal to a Platform (through the Catalog) that a service
+or plan is deprecated. The Service Broker MAY provide messaging and
+alternatives for the deprecated service or plan. This gives the Platform the
+opportunity to display meaningful advice to their users to migrate away from
+the deprecated service or plan.
+
+Service Brokers MUST allow existing Service Instances to be updated and
+bindable as they were before depreciation. Deprecation MUST NOT effect the
+usability of existing Service Instances and Bindings.
```

Comment:
> Service Brokers MUST allow existing Service Instances to be updated and bindable as they were before depreciation.

I'm concerned that situations will arise very quickly where this is not possible. Could we relax this to SHOULD?

> Deprecation MUST NOT effect the usability of existing Service Instances and Bindings.

I do, however, agree with that point.

Reply:
If the broker no longer supports the class/plan they should remove it from the catalog, and send a nasty note to their users about their dumb faces.